smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/HttpBindingProtocolGenerator.java

```diff
@@ -1685,8 +1685,23 @@ private void readDirectQueryBindings(GenerationContext context, List<HttpBinding
             String memberName = context.getSymbolProvider().toMemberName(binding.getMember());
             writer.openBlock("if (query['$L'] !== undefined) {", "}", binding.getLocationName(), () -> {
                 Shape target = context.getModel().expectShape(binding.getMember().getTarget());
-                String queryValue = getOutputValue(context, binding.getLocation(),
-                        "query['" + binding.getLocationName() + "'] as string",
+                if (target instanceof CollectionShape) {
+                    writer.write("const decoded = Array.isArray(query['$1L']) ? (query['$1L'] as string[])"
```

Comment:
nit: `$S` will put in the quotes for you, as well as handling escaping if necessary.

Reply:
yeah I don't know why I translated the previous code so literally